static/js/controllers/contacts-edit.js

```diff
@@ -192,142 +193,121 @@ angular.module('inboxControllers').controller('ContactsEditCtrl',
         .then(function(original) {
           var submitted = EnketoTranslation.contactRecordToJs(form.getDataStr());
 
-          var repeated = submitted[2];
-          var extras = submitted[1];
-          submitted = submitted[0];
-
           if(original) {
-            submitted = $.extend({}, original, submitted);
+            submitted.doc = $.extend({}, original, submitted.doc);
           } else {
-            submitted.type = $scope.enketoContact.type;
+            submitted.doc.type = $scope.enketoContact.type;
           }
 
-          return saveDoc(submitted, original, extras, repeated);
+          return prepareSubmittedDocsForSave(original, submitted);
+        })
+        .then(function(preparedDocs) {
+          return DB().bulkDocs(preparedDocs.preparedDocs)
+            .then(function(bulkDocResult) {
+              if (_.some(bulkDocResult, function(result) {
+                return !result.ok;
+              })) {
+                // REVIEWER: is this OK? What would you throw here. It doesn't
+                //           seem feasible to throw a generic error message and
+                //           then attach the result in a meaningful way.
+                throw bulkDocResult;
```

Comment:
plz take note. If you have any suggestions for how to deal with this I'd love to hear about it